packages/qb/pqb/src/queryMethods/then.ts

```diff
@@ -341,32 +342,30 @@ const then = async (
         // afterCommitHooks are executed later after transaction commit,
         // or, if we don't have transaction, they are executed intentionally after other after hooks
         if (afterCommitHooks) {
-          if (
-            trx &&
-            // when inside test transactions, push to a transaction only unless it's the outer user transaction.
-            (!trx.testTransactionCount ||
-              trx.transactionId + 1 > trx.testTransactionCount)
-          ) {
-            (trx.afterCommit ??= []).push(
-              result as unknown[],
-              q,
-              afterCommitHooks,
-            );
+          const boundHooks = afterCommitHooks.map((hook) => {
+            const boundHook = hook.bind(null, result, q);
+            if (hook.name) {
+              Object.defineProperty(boundHook, 'name', { value: hook.name });
+            }
+            return boundHook;
+          });
```

Comment:
This converts after-query hook (which is to take results and data) to a function with no parameters, to follow a generalized contract for hooks. I can't simply omit this, after-query hooks won't get the data they need.

This used to be stored as `[hook1, results1, query1, hook2, results2, query2]` (apparently some micro-optimization) and now as hooks won't necessary have results and query, I generalized the store as `[hook1, hook2, hook3]` and for this I had to bind them here.

If you want to avoid binding, the hooks can be stored as `[{hook, arguments}, {hook, arguments}]` or even `[hook1, arguments1, hook2, arguments2, ...]` having null for afterCommit hook arguments. To be honest I don't think micro-optimization matters here, it's a rarely used branch with not much data involved.

Please let me know specifically how do you anticipate the generalized hook list store.

Reply:
> apparently some micro-optimization

Yes, that's it.

> having null for afterCommit hook arguments.

But why? afterCommit hooks still expect two parameters, here you're binding them.

> To be honest I don't think micro-optimization matters here

Sure, they never matters, it's more a principle thing. If this isn't hard to micro-optimize - why not?
Especially if this micro-optimization was already there, it hurts a little bit to deoptimize.

Reply:
> afterCommit hooks still expect two parameters, here you're binding them.

It doesn't, because afterCommit can be called outside the transaction and thus there is no query and results in context.

So we should either prevent this (crash in runtime if `db.$afterCommit(hook)` is called when not in transaction) or allow both after-commit query hooks (which do receive query and data) and after-commit transactions hooks (which don't receive query and data).

I believe allowing `db.$afterCommit` to work outside the transaction is more flexible, given that transactions in Orchid are quite transparent.